src/storage/MonitoringStore.ts

```diff
@@ -22,41 +22,58 @@ export class MonitoringStore<T extends ResourceStore = ResourceStore>
     this.identifierStrategy = identifierStrategy;
   }
 
+  public async getRepresentation(identifier: ResourceIdentifier, preferences: RepresentationPreferences,
+    conditions?: Conditions): Promise<Representation> {
+    return this.source.getRepresentation(identifier, preferences, conditions);
+  }
+
+  public async modifyResource(identifier: ResourceIdentifier, patch: Patch,
+    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    const created = await this.source.modifyResource(identifier, patch, conditions);
+    const containersChanged = created
+      .filter((resource): boolean => !this.identifierStrategy.isRootContainer(resource))
+      .map((resource): ResourceIdentifier => this.identifierStrategy.getParentContainer(resource));
+    this.emitChanged(identifier, containersChanged);
+    return created;
+  }
+
+  public async setRepresentation(identifier: ResourceIdentifier, representation: Representation,
+    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    const created = await this.source.setRepresentation(identifier, representation, conditions);
+    const containersChanged = created
+      .filter((resource): boolean => !this.identifierStrategy.isRootContainer(resource))
+      .map((resource): ResourceIdentifier => this.identifierStrategy.getParentContainer(resource));
+
+    this.emitChanged(identifier, containersChanged);
+    return created;
```

Comment:
Can't have repeated code like this; needs to move to `emitChanged`.

But also, if `setRepresentation` returns a list, we should not have to modify it. We should not need to call `getParentContainer` anymore. Plus a lot of potential for duplicates. Plus there is no defined order.